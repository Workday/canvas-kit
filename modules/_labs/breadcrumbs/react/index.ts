import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
