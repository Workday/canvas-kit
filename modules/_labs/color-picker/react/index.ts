export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
