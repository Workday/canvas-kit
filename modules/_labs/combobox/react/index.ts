import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
