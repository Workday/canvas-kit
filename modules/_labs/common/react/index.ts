export * from './lib/Flex';
export * from './lib/Box';
