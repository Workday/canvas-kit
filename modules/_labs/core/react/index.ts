import type from './lib/type';
import space from './lib/space';
export default type;
export {type, space};
export * from './lib/type';
export * from './lib/StaticStates';
