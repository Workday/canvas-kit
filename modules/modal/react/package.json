{
  "name": "@workday/canvas-kit-react-modal",
  "version": "3.4.0",
  "description": "Modal to display a popup with a translucent background",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "http://github.com/Workday/canvas-kit/tree/master/modules/modal/react"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts"
  ],
  "scripts": {
    "watch": "yarn build:es6 -w",
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build": "npm-run-all --parallel build:cjs build:es6",
    "depcheck": "node ../../../utils/check-dependencies-exist.js"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "modal"
  ],
  "dependencies": {
    "@emotion/core": "^10.0.21",
    "@emotion/styled": "^10.0.17",
    "@workday/canvas-kit-react-popup": "^3.4.0",
    "focus-trap-js": "1.0.8"
  },
  "peerDependencies": {
    "react": ">= 16.8 < 17"
  }
}
