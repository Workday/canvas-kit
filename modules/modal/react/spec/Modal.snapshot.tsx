import * as React from 'react';
import Modal from '../lib/Modal';
import * as renderer from 'react-test-renderer';

describe('Modal Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<Modal open={true} padding={Modal.Padding.s} />);
    expect(component).toMatchSnapshot();
  });
  test('renders Modal with close icon', () => {
    const component = renderer.create(
      <Modal open={true} handleClose={jest.fn()} padding={Modal.Padding.s} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders Modal with different padding', () => {
    const component = renderer.create(
      <Modal open={true} handleClose={jest.fn()} padding={Modal.Padding.l} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders Modal with children elements', () => {
    const component = renderer.create(
      <Modal open={true} handleClose={jest.fn()} padding={Modal.Padding.zero}>
        <span>hello world</span>
      </Modal>
    );
    expect(component).toMatchSnapshot();
  });
  test('renders Modal without close icon', () => {
    const component = renderer.create(
      <Modal open={true} padding={Modal.Padding.zero}>
        <span>hello world</span>
      </Modal>
    );
    expect(component).toMatchSnapshot();
  });
  test('renders Modal with transformOrigin', () => {
    const component = renderer.create(
      <Modal
        open={true}
        transformOrigin={{vertical: 'top', horizontal: 'center'}}
        padding={Modal.Padding.zero}
      >
        <span>hello world</span>
      </Modal>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders Modal with a different width', () => {
    const component = renderer.create(
      <Modal open={true} width={Modal.Width.m}>
        <span>hello world</span>
      </Modal>
    );
    expect(component).toMatchSnapshot();
  });
});
