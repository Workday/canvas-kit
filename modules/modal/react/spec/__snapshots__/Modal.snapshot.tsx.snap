// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Snapshots renders Modal with a different width 1`] = `
.emotion-3 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(0,0,0,0.65);
  -webkit-animation-name: animation-h2u5si;
  animation-name: animation-h2u5si;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
}

.emotion-0 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-2 {
  position: relative;
  width: 800px;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-1 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 32px;
  width: 800px;
}

<div
  className="emotion-3 emotion-4"
  onClick={[Function]}
>
  <div
    className="emotion-2"
    role="dialog"
    width="800px"
  >
    <div
      className="emotion-1"
      width="800px"
    >
      <div
        className="emotion-0"
      >
        <span>
          hello world
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders Modal with children elements 1`] = `
.emotion-7 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(0,0,0,0.65);
  -webkit-animation-name: animation-h2u5si;
  animation-name: animation-h2u5si;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
}

.emotion-6 {
  position: relative;
  width: 440px;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-4 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-3 {
  position: absolute;
  right: 12px;
  top: 12px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: transparent;
  margin: -8px;
  width: 40px;
  height: 40px;
}

.emotion-1:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-1:disabled,
.emotion-1:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-1 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-1 span .wd-icon-fill,
.emotion-1 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-1:focus {
  background-color: transparent;
}

.emotion-1:focus span .wd-icon-fill,
.emotion-1:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-1:hover:focus {
  background-color: transparent;
}

.emotion-1:active,
.emotion-1:focus:active,
.emotion-1:hover:active {
  background-color: transparent;
  border-color: transparent;
}

.emotion-1:active span .wd-icon-fill,
.emotion-1:focus:active span .wd-icon-fill,
.emotion-1:hover:active span .wd-icon-fill,
.emotion-1:active span .wd-icon-accent,
.emotion-1:focus:active span .wd-icon-accent,
.emotion-1:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:hover {
  background-color: transparent;
}

.emotion-1:hover span .wd-icon-fill,
.emotion-1:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:disabled,
.emotion-1:active:disabled,
.emotion-1:focus:disabled,
.emotion-1:hover:disabled {
  background-color: transparent;
}

.emotion-1:disabled span .wd-icon-fill,
.emotion-1:active:disabled span .wd-icon-fill,
.emotion-1:focus:disabled span .wd-icon-fill,
.emotion-1:hover:disabled span .wd-icon-fill,
.emotion-1:disabled span .wd-icon-accent,
.emotion-1:active:disabled span .wd-icon-accent,
.emotion-1:focus:disabled span .wd-icon-accent,
.emotion-1:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-1:not([disabled]):focus {
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-1:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-5 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 0px;
  width: 440px;
}

<div
  className="emotion-7 emotion-8"
  onClick={[Function]}
>
  <div
    className="emotion-6"
    role="dialog"
    width="440px"
  >
    <div
      className="emotion-3"
    >
      <button
        aria-label="Close"
        className="emotion-1 emotion-2"
        onClick={[MockFunction]}
        title="Close"
      >
        <span
          className="emotion-0"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
            }
          }
        />
      </button>
    </div>
    <div
      className="emotion-5"
      width="440px"
    >
      <div
        className="emotion-4"
      >
        <span>
          hello world
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders Modal with close icon 1`] = `
.emotion-7 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(0,0,0,0.65);
  -webkit-animation-name: animation-h2u5si;
  animation-name: animation-h2u5si;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
}

.emotion-6 {
  position: relative;
  width: 440px;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-5 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 16px;
  width: 440px;
}

.emotion-4 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-3 {
  position: absolute;
  right: 12px;
  top: 12px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: transparent;
  margin: -8px;
  width: 40px;
  height: 40px;
}

.emotion-1:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-1:disabled,
.emotion-1:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-1 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-1 span .wd-icon-fill,
.emotion-1 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-1:focus {
  background-color: transparent;
}

.emotion-1:focus span .wd-icon-fill,
.emotion-1:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-1:hover:focus {
  background-color: transparent;
}

.emotion-1:active,
.emotion-1:focus:active,
.emotion-1:hover:active {
  background-color: transparent;
  border-color: transparent;
}

.emotion-1:active span .wd-icon-fill,
.emotion-1:focus:active span .wd-icon-fill,
.emotion-1:hover:active span .wd-icon-fill,
.emotion-1:active span .wd-icon-accent,
.emotion-1:focus:active span .wd-icon-accent,
.emotion-1:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:hover {
  background-color: transparent;
}

.emotion-1:hover span .wd-icon-fill,
.emotion-1:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:disabled,
.emotion-1:active:disabled,
.emotion-1:focus:disabled,
.emotion-1:hover:disabled {
  background-color: transparent;
}

.emotion-1:disabled span .wd-icon-fill,
.emotion-1:active:disabled span .wd-icon-fill,
.emotion-1:focus:disabled span .wd-icon-fill,
.emotion-1:hover:disabled span .wd-icon-fill,
.emotion-1:disabled span .wd-icon-accent,
.emotion-1:active:disabled span .wd-icon-accent,
.emotion-1:focus:disabled span .wd-icon-accent,
.emotion-1:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-1:not([disabled]):focus {
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-1:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

<div
  className="emotion-7 emotion-8"
  onClick={[Function]}
>
  <div
    className="emotion-6"
    role="dialog"
    width="440px"
  >
    <div
      className="emotion-3"
    >
      <button
        aria-label="Close"
        className="emotion-1 emotion-2"
        onClick={[MockFunction]}
        title="Close"
      >
        <span
          className="emotion-0"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
            }
          }
        />
      </button>
    </div>
    <div
      className="emotion-5"
      width="440px"
    >
      <div
        className="emotion-4"
      />
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders Modal with different padding 1`] = `
.emotion-7 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(0,0,0,0.65);
  -webkit-animation-name: animation-h2u5si;
  animation-name: animation-h2u5si;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
}

.emotion-6 {
  position: relative;
  width: 440px;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-4 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-3 {
  position: absolute;
  right: 12px;
  top: 12px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: transparent;
  margin: -8px;
  width: 40px;
  height: 40px;
}

.emotion-1:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-1:disabled,
.emotion-1:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-1 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-1 span .wd-icon-fill,
.emotion-1 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-1:focus {
  background-color: transparent;
}

.emotion-1:focus span .wd-icon-fill,
.emotion-1:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-1:hover:focus {
  background-color: transparent;
}

.emotion-1:active,
.emotion-1:focus:active,
.emotion-1:hover:active {
  background-color: transparent;
  border-color: transparent;
}

.emotion-1:active span .wd-icon-fill,
.emotion-1:focus:active span .wd-icon-fill,
.emotion-1:hover:active span .wd-icon-fill,
.emotion-1:active span .wd-icon-accent,
.emotion-1:focus:active span .wd-icon-accent,
.emotion-1:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:hover {
  background-color: transparent;
}

.emotion-1:hover span .wd-icon-fill,
.emotion-1:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:disabled,
.emotion-1:active:disabled,
.emotion-1:focus:disabled,
.emotion-1:hover:disabled {
  background-color: transparent;
}

.emotion-1:disabled span .wd-icon-fill,
.emotion-1:active:disabled span .wd-icon-fill,
.emotion-1:focus:disabled span .wd-icon-fill,
.emotion-1:hover:disabled span .wd-icon-fill,
.emotion-1:disabled span .wd-icon-accent,
.emotion-1:active:disabled span .wd-icon-accent,
.emotion-1:focus:disabled span .wd-icon-accent,
.emotion-1:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-1:not([disabled]):focus {
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-1:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-5 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 32px;
  width: 440px;
}

<div
  className="emotion-7 emotion-8"
  onClick={[Function]}
>
  <div
    className="emotion-6"
    role="dialog"
    width="440px"
  >
    <div
      className="emotion-3"
    >
      <button
        aria-label="Close"
        className="emotion-1 emotion-2"
        onClick={[MockFunction]}
        title="Close"
      >
        <span
          className="emotion-0"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
            }
          }
        />
      </button>
    </div>
    <div
      className="emotion-5"
      width="440px"
    >
      <div
        className="emotion-4"
      />
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders Modal with transformOrigin 1`] = `
.emotion-3 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(0,0,0,0.65);
  -webkit-animation-name: animation-h2u5si;
  animation-name: animation-h2u5si;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
}

.emotion-2 {
  position: relative;
  width: 440px;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-0 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-1 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 0px;
  width: 440px;
}

<div
  className="emotion-3 emotion-4"
  onClick={[Function]}
>
  <div
    className="emotion-2"
    role="dialog"
    width="440px"
  >
    <div
      className="emotion-1"
      width="440px"
    >
      <div
        className="emotion-0"
      >
        <span>
          hello world
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders Modal without close icon 1`] = `
.emotion-3 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(0,0,0,0.65);
  -webkit-animation-name: animation-h2u5si;
  animation-name: animation-h2u5si;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
}

.emotion-2 {
  position: relative;
  width: 440px;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-0 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-1 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 0px;
  width: 440px;
}

<div
  className="emotion-3 emotion-4"
  onClick={[Function]}
>
  <div
    className="emotion-2"
    role="dialog"
    width="440px"
  >
    <div
      className="emotion-1"
      width="440px"
    >
      <div
        className="emotion-0"
      >
        <span>
          hello world
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders as expected 1`] = `
.emotion-3 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(0,0,0,0.65);
  -webkit-animation-name: animation-h2u5si;
  animation-name: animation-h2u5si;
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
}

.emotion-2 {
  position: relative;
  width: 440px;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-1 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 16px;
  width: 440px;
}

.emotion-0 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

<div
  className="emotion-3 emotion-4"
  onClick={[Function]}
>
  <div
    className="emotion-2"
    role="dialog"
    width="440px"
  >
    <div
      className="emotion-1"
      width="440px"
    >
      <div
        className="emotion-0"
      />
    </div>
  </div>
</div>
`;
